### PR TITLE
Bugfix/changes to pil application

### PR DESCRIPTION
**Pull request by joefitter**

### CODE REVIEW

File: pages/pil/create/views/index.jsx [hunk 1]
```diff
@@ -18,19 +17,12 @@ const Index = ({ establishment, pilApplication, profile }) => (
...
-                    label={<Snippet>{`${link}.title`}</Snippet>}
-                  />
-                </h2>
+                <h2><Snippet>{`${link}.title`}</Snippet></h2>
                 <p>
                   <Snippet>{`${link}.subtitle`}</Snippet>
                 </p>
-                <LinkButton type="button" href={`/e/${establishment.id}/people/${profile.id}/pil/${pilApplication.id}`}>
-                  <Snippet>pil.buttons.applyNow</Snippet>
+                <LinkButton type="button" href={`${url}?action=${link}`}>
```
Comment (lennym) [inline pages/pil/create/views/index.jsx]: If you just want "the current url with a query string" then `?foo=bar` as an href attribute should do.

File: pages/pil/exemptions/routers/exempt.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+  app.use('/', form({ schema }));
+
+  app.post('/', (req, res, next) => {
+    if (req.form.values.exempt === 'Yes') {
+      set(req.session, `${req.profileId}.skipExemptions`, false);
+      return res.redirect(req.originalUrl.concat('/modules'));
+    } else {
+      set(req.session, `${req.profileId}.skipExemptions`, true);
+      return res.redirect(req.originalUrl.replace(/\/exemptions/, ''));
```
Comment (lennym) [inline pages/pil/exemptions/routers/exempt.js]: We might want to look towards some of @lvnhmd's work with URL builders here.

File: pages/pil/create/views/index.jsx (hunk 1 above)
Comment (joefitter) [reply pages/pil/create/views/index.jsx]: clever girl

File: pages/pil/exemptions/routers/exempt.js (hunk 2 above)
Comment (joefitter) [reply pages/pil/exemptions/routers/exempt.js]: yes, I wanted to tackle that separately, it looks like there are a load of conflicts which I assume will be from the link builder work

File: pages/pil/update/views/training-data.jsx [hunk 3]
```diff
@@ -42,11 +42,7 @@ class TrainingData extends Component {
               <div className="govuk-grid-column-one-quarter">
                 <ul className="actions">
                   <li>
-                    <form method="POST" noValidate>
-                      <input type="hidden" name="trainingModuleId" value={certificate.id} />
-                      <input type="hidden" name="action" value="delete" />
-                      <button type="submit" className="link"><span><Snippet>actions.remove</Snippet></span></button>
-                    </form>
+                    <a href={`${url}/training/${certificate.id}/delete?referrer=${url}`}><Snippet>actions.remove</Snippet></a>
```
Comment (lennym) [inline pages/pil/update/views/training-data.jsx]: I don't _like_ performing data manipulation on a `GET` - especially deleting data.

Comment (joefitter) [reply pages/pil/update/views/training-data.jsx]: yeah, I know what you mean.

File: pages/pil/create/index.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    const { action } = req.query;
+
+    if (action === 'catAF') {
+      return Promise.resolve()
+        // create empty PIL
+        .then(() => req.api(`/establishment/${req.establishmentId}/profiles/${req.profileId}/pil`, { method: 'POST' }))
+        // lookup created PIL
+        .then(() => req.api(`/establishment/${req.establishmentId}/profiles/${req.profileId}`))
+        .then(({ json: { data } }) => {
+          return res.redirect(req.baseUrl.replace('create', `${data.pil.id}/edit`));
```
Comment (wheelsandcogs) [inline pages/pil/create/index.js]: We should use the route builder here no?

Comment (joefitter) [reply pages/pil/create/index.js]: yep! good spot

**Pull request merged**